lib/fabrik/Agent.js

```diff
@@ -19,8 +19,46 @@ class Agent extends HttpClient {
     this.settings = settings;
   }
 
-  get basePath() {
-    return `/v${this.settings.version}`;
+  basePath(host) {
+    const urls = [];
+    urls.push(formatUrl({
+      protocol: this.protocol,
+      hostname: host,
+      port: this.port,
+      pathname: path.posix.join(`/v1`, 'info')
+    }));
+    urls.push(formatUrl({
+      protocol: this.protocol,
+      hostname: host,
+      port: this.port,
+      pathname: 'info'
+    }));
+    return Promise
+      .any(_.map(urls, url => AGENT_CACHE[host] !== undefined ?
+        Promise.resolve(`/v${AGENT_CACHE[host].api_version}`) :
+        this
+        .request({
+          method: 'GET',
+          url: url
+        }, 200)
+        .then(res => {
+          if (_.isPlainObject(res.body)) {
+            return res.body;
+          }
+          throw new Error('Received invalid /info response', res);
+        })
+        .then(info => {
+          logger.debug(`${host} agent info - `, info);
+          let api_version = '1';
+          if (info && info.api_version &&
+            !Number.isInteger(info.api_version) &&
+            compareVersions(info.api_version, '1.3') < 0) {
```

Comment:
what is 1.3 here ?

Reply:
It can be 1.2. In 1.0 only credential feature was supported. In 1.1 all features supported returned by info API. According to the above code 1.2 also supported, that's why <1.3
Reason: By this feature available in open sourced repo, if someone implemented an agent with enhanced features and given version 1.2. To allow one grace version I increased one more minor version.